### DISCUSSION TITLE
[vcpkg-tool] Allow baseline to use a tag as well as commit sha

### DISCUSSION THREAD

**Discussion by GordonSmith** [microsoft/vcpkg#35770] [Ideas]
### Is your feature request related to a problem? Please describe.

Its frustrating when viewing `vcpkg-configuration.json` files that have a `baseline` which refers to a specific tag as we must use a commit SHA, which doesn't convey any "when" or "version" information:
```json
{
  "default-registry": {
    "kind": "git",
    "baseline": "9edb1b8e590cc086563301d735cae4b6e732d2d2",
    "repository": "https://github.com/microsoft/vcpkg"
  },
  "overlay-ports": [
    "./vcpkg-overlays"
  ],
  "overlay-triplets": [],
  "registries": []
}
```

### Proposed solution

What would be better IMO would be to allow tags:
```json
{
  "default-registry": {
    "kind": "git",
    "baseline": "2023.08.09",
    "repository": "https://github.com/microsoft/vcpkg"
  },
  "overlay-ports": [
    "./vcpkg-overlays"
  ],
  "overlay-triplets": [],
  "registries": []
}
```

### Describe alternatives you've considered

NA

**Comment by kollinmurphy** [comment]
I absolutely agree. This feature would be wonderful to have.
